src/Controller/UserController.php

```diff
@@ -35,31 +35,25 @@ public function registerAction() {
         $userName = $_POST['userName'];
         $password = $_POST['password'];
 
+
         if ($this->userExist($userName)) {
-            $error = "Užsiregistruoti nepavyko, toks vartotojo vardas jau egzistuoja";
-            include("views/error.php");
-            return;
+
+            throw new PublicException("Užsiregistruoti nepavyko, toks vartotojo vardas jau egzistuoja");

```

Comment:
Galėtume sakyt kad čia leidžiam "black hat" sužinot kad atspėjo vartotojo vardą. Na bet čia neparašant esmės suklaidintume paprastą vartotoją, todėl tenka šiek tiek aukoti saugumą

Reply:
Taip, bet šitą naudoja ir kitos svetainės kai registruojiesi, todėl čia toks "globalus" saugumo aukojimas.